packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/framework.ts

```diff
@@ -109,6 +117,23 @@ async function invokeUserFunction(functionArnEnv: string, payload: any) {
 
   const jsonPayload = parseJsonPayload(resp.Payload);
   if (resp.FunctionError) {
+    if (resp.FunctionError.includes('Lambda is initializing your function')) {
+      const newDelay = retryOptions ? retryOptions.delay * 2 : DEFAULT_DELAY;
```

Comment:
I think this is trying to do an exponential backoff, but I can't really follow it very well. A closed-form solution is this (don't forget the jitter):

```ts
// retryAttempt = 0, 1, 2, ...
const delayMs = Math.floor(BASE_SLEEP * Math.pow(2, retryAttempt) * Math.random());
```